From 95fbeff7d12ad8422667b70905f1698b1a32a57a Mon Sep 17 00:00:00 2001
From: haiqiang yao <haiqiangx.yao@intel.com>
Date: Mon, 20 Jul 2015 17:54:02 +0800
Subject: [PATCH 452/462] ov5693:fix the stretching issue for 1296x976 sensor
 setting

At present,the 1296x976 sensor setting scales from 2624x1492 to
1296x976 and it will cause preview stretch.

the patch will change the 1296x976 sensor setting scaling from
2592x1952 to 1296x976.

Change-Id: I0e0c7df3ecf844857761397e68a5b772a649575a
Tracked-On: https://jira01.devtools.intel.com/browse/GMINL-13797
Signed-off-by: haiqiang yao <haiqiangx.yao@intel.com>
Reviewed-on: https://icggerrit.ir.intel.com/65672
Build: IMIN (SC)
Approver: Jing Tao <jing.tao@intel.com>
Validation-Android: Wei Tao <wei.tao@intel.com>
Integrator: WeiX Meng <weix.meng@intel.com>
Build: WeiX Meng <weix.meng@intel.com>
Reviewed-by: WeiX Meng <weix.meng@intel.com>
Maintainer: WeiX Meng <weix.meng@intel.com>
---
 .../external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h  | 7 +++++--
 1 file changed, 5 insertions(+), 2 deletions(-)

diff --git a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
index 0772541..2cccc53 100755
--- a/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
+++ b/drivers/external_drivers/camera/drivers/media/i2c/ov5693-ecs/ov5693.h
@@ -550,7 +550,7 @@ static struct ov5693_reg const ov5693_654x496[] = {
 
 /*
  * 1296x976 30fps 17ms VBlanking 2lane 10Bit (Scaling)
-*DS from 2624x1492
+*DS from 2592x1952
 */
 static struct ov5693_reg const ov5693_1296x976[] = {
 	{OV5693_8BIT, 0x3501, 0x7b},
@@ -578,7 +578,10 @@ static struct ov5693_reg const ov5693_1296x976[] = {
 	{OV5693_8BIT, 0x380e, 0x07},
 	{OV5693_8BIT, 0x380f, 0xc0},
 
-	{OV5693_8BIT, 0x3813, 0xE8},
+	{OV5693_8BIT, 0x3810, 0x00},
+	{OV5693_8BIT, 0x3811, 0x10},
+	{OV5693_8BIT, 0x3812, 0x00},
+	{OV5693_8BIT, 0x3813, 0x02},
 
 	{OV5693_8BIT, 0x3814, 0x11},	/*X subsample control*/
 	{OV5693_8BIT, 0x3815, 0x11},	/*Y subsample control*/
-- 
1.9.1

